scaper/audio.py

```diff
@@ -8,56 +8,17 @@
 import subprocess
 import sox
 import numpy as np
+import pyloudnorm
 import soundfile
 import tempfile
 from .scaper_exceptions import ScaperError
 from .util import _close_temp_files
 
-
-def r128stats(filepath):
-    """ takes a path to an audio file, returns a dict with the loudness
-    stats computed by the ffmpeg ebur128 filter """
-    ffargs = ['ffmpeg',
-              '-nostats',
-              '-i',
-              filepath,
-              '-filter_complex',
-              'ebur128',
-              '-f',
-              'null',
-              '-']
-    try:
-        proc = subprocess.Popen(ffargs, stderr=subprocess.PIPE,
-                                universal_newlines=True)
-        stats = proc.communicate()[1]
-        summary_index = stats.rfind('Summary:')
-
-        if summary_index == -1:
-            raise ScaperError(
-                'Unable to find LUFS summary, stats string:\n{:s}'.format(
-                    stats))
-
-        summary_list = stats[summary_index:].split()
-        i_lufs = float(summary_list[summary_list.index('I:') + 1])
-        i_thresh = float(summary_list[summary_list.index('I:') + 4])
-        lra = float(summary_list[summary_list.index('LRA:') + 1])
-        lra_thresh = float(summary_list[summary_list.index('LRA:') + 4])
-        lra_low = float(summary_list[summary_list.index('low:') + 1])
-        lra_high = float(summary_list[summary_list.index('high:') + 1])
-        stats_dict = {'I': i_lufs, 'I Threshold': i_thresh, 'LRA': lra,
-                      'LRA Threshold': lra_thresh, 'LRA Low': lra_low,
-                      'LRA High': lra_high}
-    except Exception as e:
-        raise ScaperError(
-            'Unable to obtain LUFS data for {:s}, error message:\n{:s}'.format(
-                filepath, e.__str__()))
-
-    return stats_dict
-
-
-def get_integrated_lufs(filepath, min_duration=0.5):
+def get_integrated_lufs(audio_array, samplerate, min_duration=0.5,
+                        filter_class='K-weighting', block_size=0.400):
     """
-    Returns the integrated LUFS for an audiofile.
+    Returns the integrated LUFS for a numpy array containing
+    audio samples.
 
     For files shorter than 400 ms ffmpeg returns a constant integrated LUFS
```

Comment:
Since we're not using ffmpeg anymore we should update this comment. We could still keep the min duration, since perhaps pyloudnorm will also become unstable for very short audio clips